### PR TITLE
[4] PHP 8.2 Dynamic property - languagefilter

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue php 8.2 - dynamic property Deprecated

.### Summary of Changes
removed  Creation of dynamic property `item_associations`

### Testing Instructions
open frontend with php 8.2 with Error Reporting = maximum and multilaguage enabled

### Actual result BEFORE applying this Pull Request
Deprecated: Creation of dynamic property Joomla\CMS\Application\SiteApplication::$item_associations is deprecated

### Expected result AFTER applying this Pull Request

no more

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 93d7c00af962c277a422a1f74956fe83b8751ff0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39630">issues.joomla.org/tracker/joomla-cms/39630</a>.</sub>

**Comment by HLeithner** [comment]
It's a b/c break isn't it?

**Comment by alikon** [comment]
are the plugin covered by b/c policies ?

**Comment by HLeithner** [comment]
It's similar to https://github.com/joomla/joomla-cms/pull/39592 no idea if it has the same impact but we don't need to do breaks if it can be avoided and cleaned up in 5.0

**Comment by HLeithner** [comment]
That's the reason I said similar and not the same.
Anyway any research don if a extension is using it?

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 93d7c00af962c277a422a1f74956fe83b8751ff0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39630">issues.joomla.org/tracker/joomla-cms/39630</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39630">issues.joomla.org/tracker/joomla-cms/39630</a>.</sub>

**Comment by laoneo** [comment]
@alikon can you add the variable to the Application class so we can add a deprecation message in 4.3? Added a new label "Needs deprecation" so we will not forget.

**Comment by alikon** [comment]
done

**Comment by laoneo** [comment]
You need to remove the deprecation as we are not allowed to deprecate stuff in a patch release. But I gave it a proper label, so we can deprecate in 4.3 then.

**Comment by alikon** [comment]
ok done

File: libraries/src/Application/WebApplication.php [hunk 1]
```diff
@@ -45,6 +45,14 @@ abstract class WebApplication extends AbstractWebApplication
      */
     public $JComponentTitle;
 
+    /**
+     * item associations
```
Comment (laoneo) [inline libraries/src/Application/WebApplication.php]:
```suggestion
     * The item associations
```

**Comment by fancyFranci** [comment]
Thank you!

**Pull request merged**